### ISSUE TITLE
[codelab-intro-mnist-feedback]:

### ISSUE THREAD

**Issue by crispindev** [kubeflow/examples#701]
Hi Team,

I tried running the mnist-demo. As soon as I try to run kustomize build . | kubectl apply -f -

I get:
configmap/mnist-map-training-g9gg26tmm2 created
tfjob.kubeflow.org/my-train-1 created

When describing the tfjob, I keep getting the same error:
Warning  SettedPodTemplateRestartPolicy  4m58s (x13 over 5m10s)  tf-operator  Restart policy in pod template will be overwritten by restart policy in replica spec
Warning  FailedCreatePod                 4m58s (x12 over 5m9s)   tf-operator  Error creating: pods "my-train-1-chief-0" is forbidden: error looking up service account kubeflow/default-editor: serviceaccount "default-editor" not found

Any ideas on why that might happen?

**Comment by zhaoxq** [comment]
I see same error. I tried to create a service account with name "default-editor", I still see the same error.

**Comment by cbscientist** [comment]
I see the same error as well. My guess is that the default service account for the kubeflow deployment underwent a name change and the corresponding .yaml files (6 total under the mnist-deployment directory) have yet to be updated.

**Comment by cbscientist** [comment]
OK - brief follow up here. I'm still working through some other issues with this example but in this case the service account referred to in the error message refers to a Kubernetes serviceAccount object, not a GCP service account which I had previously assumed. Listing all service accounts via the `kubectl get serviceAccounts` command provided a list of names to choose from. I went with the `default` k8s service account, updated the YAML files accordingly, and redeployed my training job. This resolved the error initially described in this thread, but I am still receiving permission errors from TF when it attempts to make an HTTP request to one of the compute engine vms. Working through this now.

**Comment by sanjikasehgal** [comment]
Hey I'm getting that same issues as well with the HTTP request permission error. I've tried giving admin level permissions to the service accounts at both the object and bucket level. Did you ever resolve that issue? @cbscientist

**Comment by cbscientist** [comment]
@sanjikasehgal Kind of? I ended up making the following changes, which helped narrow down the issue (this was after I fixed the reference to the K8s service account and added write privileges to the GCS service accounts, but looking back I'm not sure I needed to do the latter).

1. In `training/base/Worker.yaml`, I commented out the command to run `model.py` and instead put in some code to just run the container indefinitely:

```
- sleep
- infinity
```

2. I then used the process documented [here](https://kubernetes.io/docs/tasks/debug-application-cluster/get-shell-running-container/) to open up a shell into the running container and manually checked the environment variables, permissions, & settings. It was clear that whatever was supposed to happen with the authentication component, didn't.

3. I manually set the environmental references to the K8s secret in the running container, then ran `python opt/model.py` with the appropriate flags. The model was successfully trained and output to a binary file on GCS.

I ended up moving on to other work, but looking back I was following the codelab tutorial, not the README, which I think might have been a mistake. It looks like the authentication process in the README is different and uses GCP Workload Identities instead of explicitly setting a reference to a K8s secret. If you haven't already I would maybe try that and see if you get a different result.
